Query Browser: Fix bug that caused graph to re-render when typing

kyoto opened the pull request:
Move memoize logic up to the QueryBrowserPage component. This fixes a
bug where re-rendering the graph was triggered unnecessarily, including
when you typed into a query input.

TheRealJon commented on frontend/public/components/monitoring/metrics.tsx:
```diff
@@ -453,7 +453,13 @@ export const QueryBrowserPage = withFallback(() => {
   let colorOffset = 0;
 
-  const queryStrings = _.map(queries, 'query');
+  const queriesMemoKey = JSON.stringify(_.map(queries, 'query'));
```
We are doing the same (maybe even a more expensive) calculation here to create a memo key. We might as well just take the `React.useMemo` statements out and calculate the values on every render, since we are doing the same operation to create a key.

kyoto replied:
Sorry, I should have explained better. I'm using `React.useMemo` to prevent re-renders rather than to save time calculating these values. They are passed as props to child components and if they are recalculated on every render, they will always be new arrays, causing child components to re-render. Normally, this wouldn't be an issue, but in this case the graph is a child component and is slow to re-render.

TheRealJon replied:
Hmm, okay. That makes sense. Could we add a comment somewhere in here to explain that?

kyoto replied:
👍 Added

spadgett commented:
/kind bug

spadgett commented:
/retest

kyoto commented:
Rebased and added comments as suggested by https://github.com/openshift/console/pull/2274#discussion_r311581794 and https://github.com/openshift/console/pull/2245#issuecomment-519106098.

spadgett commented:
/retest

kyoto commented:
/retest
